### ISSUE TITLE
no-undef-class between version 2.3.0 -> 2.4.0

### ISSUE THREAD

**Issue by gdub22** [atfzl/eslint-plugin-css-modules#10]
I've been using this plugin for a while (thanks!) and since upgrade to 2.4.0, I now have a ton of `no-undef-class` errors that were previously valid.

I will investigate more and follow up, but maybe you know?

**Comment by atfzl** [comment]
The main change was regarding issue #4 (support for `composes` and `@extend` keyword).

Are the new errors valid errors or bugs ?

**Comment by gdub22** [comment]
Seems to be a bug because the classes are defined.  I found that the errors appear on definitions like this. It says `item, selected, open, menu, active` are all undefined.
```css
.item.selected { }
.open .menu { }
a.active { }
.menu { } /* even this reports as undefined because it was previously grouped with .open */
.foo { } /* this is okay because it wasnt used elsewhere */
```


I am also now seeing errors when I import more than 1 stylesheet into a component, it says all the styles referenced from one of them are undefined.  Possibly related.
```js
import style from './style.css'
import themeStyle from './red-theme.css'
```

**Comment by atfzl** [comment]
Can you provide a complete js and css file for reproducing this.
I am not able to replicate these in my setup.

Thanks for reporting the issue 👍

**Comment by atfzl** [comment]
I just tried these:

js file
```js
import React from 'react';

import s from './testFile.scss';
import s2 from './testFile2.scss';

export const Foo = () => (
  <div>
    <div className={s.item}>
      <div className={s.selected}></div>
    </div>
    <div className={s.open}>
      <a className={s.active}>link</a>
      <div className={s.menu}></div>
    </div>
    <div className={s.menu}>
    </div>
    <div className={s2.bar}>
    </div>
  </div>
);
```

scss file 1:
```scss
.item.selected {}
.open .menu {}
a.active {}
.menu {}
.foo {}
```

scss file 2:
```scss
.bar {}
```

It is giving only one error: `Unused classes found: foo [css-modules/no-unused-class]`
which is expected.

**Comment by gdub22** [comment]
Ok found a minimal reproduction. It happens whenever there is a `:root{}` variable block in the css.  If I remove that, it works.

js:
```js
import style from './style.css'

export const Foo = () => (
  <div class={ style.foo }></div>
)
```

style.css:
```css
:root {
  --color: red;
}

.foo { }
```

`error  Class 'foo' not found  css-modules/no-undef-class`

**Comment by atfzl** [comment]
I've found the issue. Will update with fix.

**Comment by atfzl** [comment]
I have reverted one change and it should work as it was working before in 2.3.0.

But the real issue is something else.
Gonzales-pe is not able to parse `--color`  in `:root` block and was failing.
Will open an issue regarding this and fix it later on.

**Comment by gdub22** [comment]
Thank you for the quick fix.  Confirmed it is back to the 2.3.0 behavior.
